src/main/java/org/usfirst/frc4904/robot/subsystems/Flywheel.java

```diff
@@ -0,0 +1,89 @@
+package org.usfirst.frc4904.robot.subsystems;
+
+import edu.wpi.first.wpilibj.SpeedController;
+
+import org.usfirst.frc4904.standard.custom.motioncontrollers.MotionController;
+import org.usfirst.frc4904.standard.subsystems.motor.VelocitySensorMotor;
+import org.usfirst.frc4904.standard.subsystems.motor.speedmodifiers.SpeedModifier;
+
+/**
+ * Flywheel Base Class
+ * 
+ * ! This class does not include a piston, use Shooter.java for the 2020-season
+ * ! specific flywheel subsystem.
+ * 
+ * This class is designed to go into standard, eventually.
+ */
+public class Flywheel extends VelocitySensorMotor {
+  public static enum FlywheelStatus {
+    IDLE, SPINNING_UP, AT_SPEED
+  }
+
+  protected FlywheelStatus currentStatus = FlywheelStatus.IDLE;
+  protected double targetSpeed = 0.0;
+
+  /**
+   * Flywheel constructor - extends VelocitySensorMotor
+   * motionController.absoluteTolerance should be set.
+   * 
+   * @param name             The name of the flywheel
+   * @param SpeedModifier    The speedModifier
+   * @param motionController The motionController
+   * @param motors           Motors to control
+   */
+  public Flywheel(String name, SpeedModifier speedModifier, MotionController motionController,
+      SpeedController... motors) {
+    super(name, speedModifier, motionController, motors);
+    super.enableMotionController();
+  }
+
+  /**
+   * Flywheel constructor - extends VelocitySensorMotor
+   * 
+   * @param SpeedModifier    The speedModifier
+   * @param motionController The motionController
+   * @param motors           Motors to control
+   */
+  public Flywheel(SpeedModifier speedModifier, MotionController motionController, SpeedController... motors) {
+    this("Flywheel", speedModifier, motionController, motors);
+  }
+
+  protected void syncStatus() {
+    if (Math.abs(targetSpeed) < motionController.getAbsoluteTolerance() || currentStatus == FlywheelStatus.IDLE) {
```

Comment:
did you forget to subtract the current speed from the target speed?

Reply:
I also think that we should be using our own tolerance value that is greater than `Double.MIN_VALUE`.

Reply:
Yes, I forgot to subtract.
I assume that the value will get changed in standard, I think Nikhil is working on that.

Reply:
NVM, that math is actually correct... it checks if the `targetSpeed` is around zero and if so sets the `flywheelStatus` to `IDLE`. I would've been very sad if I hadn't caught that myself...

We talked about using the `motionController` tolerance before, and decided that it should go in `robotMap`. What value should I set it to? Or should I use a `static final` in the `Flywheel` class itself? Pinging @NikhilSuresh24 as well because this is an OOP line drawing kind of decision. I suggest we talk about this in detail tomorrow, but in general I see two options:
1. Leave it the way it is, and define the constant in `robotMap`
2. Define this constant in the `Flywheel` class, although I don't like this because that constant is probably specific to our robot and not a generic flywheel. Because this file is destined for standard, I don't think we should use this option.

Reply:
I think we should put it in constants in `RobotMap` and pass it into the flywheel, but we can discuss